### PR TITLE
fix(recorder): fix recorder injected icons

### LINKED ISSUE #33194: [Bug]: icons missing in recorder dialog when inside iframe
### Version

1.48.1

### Steps to reproduce

1. Run:

```
npx playwright codegen "data:text/html,<iframe style='width:100%;height:100%' srcdoc='<h1>Hello World</h1>'></iframe>"
```

2. Use **Assert Text** to pick `Hello World`

### Expected behavior

Dialog should display accept and close icons.

### Actual behavior

Dialog accept and close buttons appear like this:

Image

### Additional context

Icons are svg `clipPaths` that are only attached to `x-pw-overlay`, which is not rendered on iframes. I can submit a PR to fix it.

### Environment

  System:
    OS: Windows 11 10.0.22631
    CPU: (12) x64 13th Gen Intel(R) Core(TM) i5-1335U
    Memory: 10.95 GB / 31.66 GB
  Binaries:
    Node: 20.17.0 - ~\scoop\apps\nvm\current\nodejs\nodejs\node.EXE
    npm: 10.8.2 - ~\scoop\apps\nvm\current\nodejs\nodejs\npm.CMD
  IDEs:
    VSCode: 1.94.2 - C:\Users\rui.figueira\AppData\Local\Programs\Microsoft VS Code\bin\code.CMD
  Languages:
    Bash: 5.2.37 - C:\Users\rui.figueira\scoop\shims\bash.EXE
  npmPackages:
    @playwright/test: ^1.48.1 => 1.48.1

### PR DESCRIPTION
**Pull request by ruifigueira**
Proposal PR to fix #33194

### CODE REVIEW

File: packages/playwright-core/src/server/injected/recorder/recorder.ts [hunk 1]
```diff
@@ -1094,6 +1093,7 @@ export class Recorder {
     recreationInterval = this.injectedScript.builtinSetTimeout(recreate, 500);
     this._listeners.push(() => clearInterval(recreationInterval));
 
+    this.highlight.appendChild(createSvgElement(this.document, clipPaths));
```
Comment (Skn0tt) [inline packages/playwright-core/src/server/injected/recorder/recorder.ts:1096]: This change is a little scary to me. Let me check if the SVG icons are used anywhere outside the `highlight` element.

Comment (Skn0tt) [reply packages/playwright-core/src/server/injected/recorder/recorder.ts:1096]: All mentions of `url(#icon-...` are in `highlight.css`, so this should be fine.

**Review by Skn0tt — APPROVED** [review]
Looks good to me. Thanks for the contribution!

**Comment by ruifigueira** [comment]
> Could you take a look at the failing test? The new `svg` element is making a locator fail because there's now two `svg` elements.

Fixed.

**Pull request merged**